test/test_sigridci_runner.py

```diff
@@ -520,7 +520,22 @@ def testFallbackToDefaultTargetIfNoSigridObjectives(self):
         target = runner.loadSigridTarget(apiClient)
         
         self.assertEqual(target, 3.5)
-        
+
+    def testUploadShouldBeDeletedAfterSubmission(self):
+        tempDir = tempfile.mkdtemp()
+        self.createTempFile(tempDir, "a.py", "print(123)")
+
+        options = UploadOptions(sourceDir=tempDir)
+        target = TargetQuality("/tmp/nonexistent", 3.5)
+        apiClient = MockApiClient(publish=False)
+
+        runner = SigridCiRunner()
+        runner.run(apiClient, options, target, [])
+
+        for root, dirs, files in os.walk("."):
+            for file in files:
+                self.assertFalse(file.endswith(".zip"))
```

Comment:
As with the similar test in the previous attempt, this of course doesn't actually check the behaviour of the temporary directory (which we also shouldn't be testing in our code), but it does document and check the behaviour that we do not create a persisting zip file.